File: system-tests/sut-uses-logback-test/src/test/java/com/teamscale/client/SutUsesLogbackTest.java
package com.teamscale.client;

import org.conqat.lib.commons.io.ProcessUtils;
import org.junit.jupiter.api.Test;

import java.nio.file.Path;
import java.nio.file.Paths;

import static org.assertj.core.api.Assertions.assertThat;

public class SutUsesLogbackTest {

	private static final String AGENT_JAR = System.getProperty("agentJar");

	@Test
	public void systemTest() throws Exception {
		ProcessUtils.ExecutionResult result = ProcessUtils.execute(
				new String[]{"java", "-javaagent:" + AGENT_JAR + "=debug=true", "-jar", "build/libs/app.jar"});

		assertThat(result.getStdout()).contains("This warning is to test logging in the SUT");
		assertThat(result.getStdout()).doesNotContainIgnoringCase("error");
		assertThat(result.getStderr()).isEmpty();

		Path appLogFile = Paths.get("logTest/app.log");
		assertThat(appLogFile).exists();
		assertThat(appLogFile).content().contains("This warning is to test logging in the SUT");
	}

}

File: system-tests/sut-uses-logback-test/src/test/kotlin/com/teamscale/client/SutUsesLogbackTest.kt
package com.teamscale.client

import org.assertj.core.api.Assertions.assertThat
import org.conqat.lib.commons.io.ProcessUtils
import org.junit.jupiter.api.Test
import java.nio.file.Paths

class SutUsesLogbackTest {
	@Test
	@Throws(Exception::class)
	fun systemTest() {
		val result = ProcessUtils.execute(
			arrayOf("java", "-javaagent:$AGENT_JAR=debug=true", "-jar", "build/libs/app.jar")
		)

		assertThat(result.stdout).contains("This warning is to test logging in the SUT")
		assertThat(result.stdout).doesNotContainIgnoringCase("error")
		assertThat(result.stderr).isEmpty()

		val appLogFile = Paths.get("logTest/app.log")
		assertThat(appLogFile).exists()
		assertThat(appLogFile).content().contains("This warning is to test logging in the SUT")
	}

	companion object {
		private val AGENT_JAR = System.getProperty("agentJar")
	}
}
